python/link_prediction.py

```diff
@@ -0,0 +1,970 @@
+    for i in range(k):
+        score, i, recommendation = heappop(results)
+        recommendations.append(mgp.Record(score=-score, recommendation=recommendation))
+
+    return recommendations
+
+@mgp.read_proc
+def test(ctx: mgp.ProcCtx, src_vertices: List[int], dest_vertices: List[int]) -> mgp.Record(scores=mgp.Any, accuracy=mgp.Number):
+    """Offers completely the same functionality as predict.
+    Instead of receiving references to the source and destination vertex, it received list of source and destination vertices.
+    Closed for public.
+
+    Args:
+        ctx (mgp.ProcCtx): A reference to the context execution.
+        src_vertices (List[int]): Source vertices for every edge. Based on Memgraph's id property.
+        dest_vertices (List[int]): Dest vertices for every edge. Based on Memgraph's id property.
+
+    Returns:
+        Scores for every edge and final accuracy.
+    """
+
+    scores = []
+
+    for i in range(len(src_vertices)):
+        src_vertex = ctx.graph.get_vertex_by_id(src_vertices[i])
+        dest_vertex = ctx.graph.get_vertex_by_id(dest_vertices[i])
+        record = predict(ctx, src_vertex, dest_vertex)
+        scores.append(record.fields["score"])
+
+    acc = torch.sum(torch.tensor(scores, requires_grad=False) > 0.5).item() / len(scores)
+
+    return mgp.Record(scores=scores, accuracy=acc)
+
+@mgp.read_proc
+def get_training_results(ctx: mgp.ProcCtx,) -> mgp.Record(training_results=mgp.Any, validation_results=mgp.Any):
+
+    """This method is used when user wants to get performance data obtained from the last training. It is in the form of list of records where each record is a Dict[metric_name, metric_value]. Training and validation
+    results are returned.
+
+    Args:
+        ctx (mgp.ProcCtx): Reference to the context execution
+
+    Returns:
+        mgp.Record[List[LinkPredictionOutputResult]]: A list of results. If the train method wasn't called yet, it returns empty lists.
+    """
+    global training_results, validation_results
+
+    return mgp.Record(training_results=training_results, validation_results=validation_results)
+
+@mgp.read_proc
+def load_context(ctx: mgp.ProcCtx, path: str = link_prediction_parameters.context_save_dir) -> mgp.Record(status=mgp.Any):
+    """Loads torch model from given path. If the path doesn't exist, underlying exception is thrown.
+    If the path argument is not given, it loads from the default path. If the user has changed path and the context was deleted 
+    then he/she needs to send that parameter here.
+
+    Args:
+        ctx (mgp.ProcCtx): A reference to the context execution.
+
+    Returns:
+        status(mgp.Any): True just to indicate that loading went well.
+    """
+
+    global model, predictor
+    model = torch.load(path + MODEL_NAME)
+    predictor = torch.load(path + PREDICTOR_NAME)
+    return mgp.Record(status=True)
+
+##############################
+# Private helper methods.
+##############################
```

Comment:
Move it to utility

Reply:
I think that methods that are used for DGL graph creation and conversion_to_dgl_test should stay in the main module because they are dependent on the mgp context.